src/target/stm32f1.c

```diff
@@ -473,11 +588,33 @@ bool at32f40x_probe(target_s *target)
 	const uint32_t idcode = target_mem32_read32(target, STM32F1_DBGMCU_IDCODE);
 	const uint32_t series = idcode & AT32F4x_IDCODE_SERIES_MASK;
 	const uint16_t part_id = idcode & AT32F4x_IDCODE_PART_MASK;
+	// ... and first byte of UID
+	const uint32_t project_id = target_mem32_read8(target, AT32F4x_PROJECT_ID);
+
+	if (series == AT32F40_SERIES) {
+		if (project_id == 2U) // F403
+			return false;
+		if (project_id == 7U || project_id == 8U) // F403A, F407
+			return at32f40_detect(target, part_id);
+	}
+	if (series == AT32F41_SERIES) {
+		if (project_id == 4U)
+			return at32f413_detect(target, part_id);
+		if (project_id == 5U)
+			return at32f415_detect(target, part_id);
+	}
+	// Value line
+	if ((series == AT32F421_SERIES || series == AT32F421_SERIES_16KB) && (project_id == 9U))
+		return at32f421_detect(target, part_id);
+
+	if (series == AT32F425_SERIES && project_id == 0x0fU)
+		return at32f425_detect(target, part_id);
+	if ((series == AT32F423_SERIES_256KB || series == AT32F423_SERIES_128KB || series == AT32F423_SERIES_64KB) &&
+		(project_id == 0x12U))
```

Comment:
The extra parens here are superfluous and can be dropped - same on line 607 for the clause after the `&&`. They suppress the GCC warning about accidental assignment that we want here.

Reply:
Removed two pairs of parens for project_id.